docs/developers/developer-reference/rust-testing-framework.md

```diff
@@ -0,0 +1,456 @@
+
+```rust
+fn setup_crowdfunding<CrowdfundingObjBuilder>(
+    cf_builder: CrowdfundingObjBuilder,
+) -> CrowdfundingSetup<CrowdfundingObjBuilder>
+where
+    CrowdfundingObjBuilder:
+        'static + Copy + Fn(DebugApi) -> crowdfunding_esdt::ContractObj<DebugApi>,
+{
+    let rust_zero = rust_biguint!(0u64);
+    let mut blockchain_wrapper = BlockchainStateWrapper::new();
+    let owner_address = blockchain_wrapper.create_user_account(&rust_zero);
+    let first_user_address = blockchain_wrapper.create_user_account(&rust_zero);
+    let second_user_address = blockchain_wrapper.create_user_account(&rust_zero);
+    let cf_wrapper = blockchain_wrapper.create_sc_account(
+        &rust_zero,
+        Some(&owner_address),
+        cf_builder,
+        WASM_PATH,
+    );
+
+    blockchain_wrapper.set_esdt_balance(&first_user_address, CF_TOKEN_ID, &rust_biguint!(1_000));
+    blockchain_wrapper.set_esdt_balance(&second_user_address, CF_TOKEN_ID, &rust_biguint!(1_000));
+
+    blockchain_wrapper.execute_tx(&owner_address, &cf_wrapper, &rust_zero, |sc| {
+        let target = managed_biguint!(2_000);
+        let token_id = managed_token_id!(CF_TOKEN_ID);
+
+        let result = sc.init(target, CF_DEADLINE, token_id);
+        assert_eq!(result, SCResult::Ok(()));
+
+        StateChange::Commit
+    });
+    blockchain_wrapper.add_mandos_set_account(cf_wrapper.address_ref());
+
+    CrowdfundingSetup {
+        blockchain_wrapper,
+        owner_address,
+        first_user_address,
+        second_user_address,
+        cf_wrapper,
+    }
+}
+```
+
+The main object you're going to be interacting with is the `BlockchainStateWrapper`. It basically holds the entire (mocked) blockchain state at any given moment, and allows you to interact with the accounts.  
+
+As you can see in the above test, we use the said wrapper to create an owner account, two user accounts, and the Crowdfunding smart contract account.  
+
+Then, we set the ESDT balances for the two users, and deploy the smart contract, by using the `execute_tx` function of the `BlockchainStateWrapper` object. The arguments are:
+- caller address
+- contract wrapper (which contains the contract address and the contract object builder)
+- EGLD payment amount
+- a lambda function, which contains the actual execution
+
+Since this is a SC deploy, we call the `init` function. Since the contract works with managed objects, we can't use the built-in Rust BigUint. We have to use the one from elrond_wasm. To create managed types, we use the `managed_` functions. Alternatively, you can create those objects by:
+
+```rust
+let target = BigUint::<DebugApi>::from(2_000u32);
+```
+
+Similarly for other managed types.  
+
+The return type for the `execute_tx` function is an enum of two possible values: Commit and Revert. This is needed for cases when you want to check SCError cases (since the lambda can contain anything, the testing framework has no idea if your mocked transaction was successful or not).  
```

Comment:
The return type of the _lambda_ passed to `execute_tx`, not the return type of `execute_tx` itself.

Reply:
Reworded.